src/app/modules/sgid/sgid.util.ts

```diff
@@ -0,0 +1,63 @@
+import { err, ok, Result } from 'neverthrow'
+
+import { hasProp } from '../../../shared/util/has-prop'
+import {
+  AuthType,
+  BasicField,
+  IFormSchema,
+  SgidFieldTitle,
+} from '../../../types'
+import { AuthTypeMismatchError } from '../form/form.errors'
+import { ProcessedSingleAnswerResponse } from '../submission/submission.types'
+
+import { SgidForm } from './sgid.types'
+
+/**
+ * Validates that a form is an sgID form
+ * @param form Form to validate
+ */
+export const validateSgidForm = <T extends IFormSchema>(
+  form: T,
+): Result<SgidForm<T>, AuthTypeMismatchError> => {
+  return isSgidForm(form)
+    ? ok(form)
+    : err(new AuthTypeMismatchError(AuthType.SGID, form.authType))
+}
+
+// Typeguard to ensure that form has eserviceId and correct authType
+const isSgidForm = <F extends IFormSchema>(form: F): form is SgidForm<F> => {
```

Comment:
the typeguard comment seems to insinuate that this checks for the existence of an eserviceId too. Should that be part of the checks in this typeguard?

EDIT: just saw that SGID doesn't have an esrvcId. Update the comment then?